src/economy/shops/taxes.ts

```diff
@@ -0,0 +1,228 @@
+  );
+  if (transmission) {
+    transmission.amount += amount;
+  } else {
+    arr.push({
+      currency: {
+        model: currency.model,
+        unitPlural: currency.unitPlural,
+        subunitPlural: currency.subunitPlural,
+      },
+      amount: amount,
+    });
+  }
+  view.untransferred = arr;
+}
+
+/**
+ * Create a new tax chest
+ */
+registerEvent(SignChangeEvent, async (event) => {
+  const sign = event.block;
+  if (!canBecomeTaxChest(sign, event.lines)) return;
+
+  // Get the chest where the taxes will be stored
+  const chest = getBlockBehind(sign)?.state;
+  if (!(chest instanceof Chest)) return;
+
+  // Use the chest data to store the
+  const chestDataView = dataView(TaxChestData, chest);
+  chestDataView.taxCollector = event.player.uniqueId.toString();
+  chest.update();
+
+  // Store the location of the tax chest to the player's data
+  const playerDataView = dataView(TaxCollectorData, event.player);
+  playerDataView.chestLocation.worldId = chest.location.world.uID.toString();
+  playerDataView.chestLocation.x = chest.location.x;
+  playerDataView.chestLocation.y = chest.location.y;
+  playerDataView.chestLocation.z = chest.location.z;
+
+  event.player.sendMessage('Veroarkku luotu!');
+});
+
+/**
+ * List of accepted words for tax chest
+ */
+const taxWords = new Set([
+  'tax',
+  'taxes',
+  'taxchest',
+  'tax chest',
+  'vero',
+  'veroarkku',
+  'verot',
+]);
+
+/**
+ * Check if the sign and the block (chest) behind it can become a tax chest
+ * @param sign Sign in front of the tax chest
+ * @param lines Lines of the sign (From SignChangeEvent)
+ * @returns Boolean (Can the chest become a tax chest or not)
+ */
+function canBecomeTaxChest(sign: Block, lines: string[]) {
+  if (!sign) return false;
+  if (!(sign.state instanceof Sign)) return false;
+
+  // Check if lines are either "tax", "veroarkku", "tax chest" etc or empty
+  for (const line of lines) {
+    if (line && !taxWords.has(line.toLowerCase())) return false;
+  }
+  if (lines.every((line) => !line)) return false;
+
+  if (!(sign.blockData instanceof WallSign)) return false;
+  const attachedTo = getBlockBehind(sign);
+  if (!attachedTo) return false;
+  if (!(attachedTo.state instanceof Chest)) return false;
+  return true;
+}
+
+/**
+ * Add untransferred taxes to the tax chest
+ */
+registerEvent(PlayerInteractEvent, (event) => {
+  if (event.action !== Action.RIGHT_CLICK_BLOCK) return;
+  if (!(event.clickedBlock?.state instanceof Chest)) return;
+  const chest = event.clickedBlock.state;
+  const taxChestView = dataView(TaxChestData, chest);
+  if (!taxChestView.taxCollector) return;
+  const player = event.player;
+
+  const collector = Bukkit.getOfflinePlayer(
+    UUID.fromString(taxChestView.taxCollector),
+  );
+  const collectorView = dataView(TaxCollectorData, collector);
+
+  const yupLoc = collectorView.chestLocation;
+  const location = yupLocToLoc(yupLoc);
+  if (!location) return;
+
+  if (distanceBetween(location, chest.location)) {
```

Comment:
Due to floating point inaccuracies, it is *possible* this check might fail. More importantly, don't rely on zero being falsy.